### PR TITLE
[ticket/14366] Add core events to the function decode_message()

### PR DESCRIPTION
**Pull request by rxu**
Add core event to the function decode_message() in
includes/functions_content.php to allow extensions performing additional
message handling before/after decoding.

<a href="https://tracker.phpbb.com/browse/PHPBB3-14366">PHPBB3-14366</a>.

### CODE REVIEW

**Comment by rxu** [comment]
!set Event

File: phpBB/includes/functions_content.php [hunk 1]
```diff
@@ -405,12 +405,38 @@ function decode_message(&$message, $bbcode_uid = '')
...
+	* Use this event to modify the message after it is decoded
+	*
+	* @event core.decode_message_after
+	* @var string	message			The message content
+	* @var string	bbcode_uid		The message BBCode UID
+	* @var string	match			Match pattern to replace
+	* @var int		replace			Replacement for the matched text
+	* @since 3.1.8-RC1
+	*/
+	$vars = array('message', 'bbcode_uid', 'match', 'replace');
```
Comment (marc1706) [inline phpBB/includes/functions_content.php]: Same as above. Also, do you really need the match and replace here?

File: phpBB/includes/functions_content.php [hunk 2]
```diff
@@ -405,12 +405,38 @@ function decode_message(&$message, $bbcode_uid = '')
...
+	* Use this event to modify the message before it is decoded
+	*
+	* @event core.decode_message_before
+	* @var string	message			The message content
+	* @var string	bbcode_uid		The message BBCode UID
+	* @var string	match			Match pattern to replace
+	* @var int		replace			Replacement for the matched text
+	* @since 3.1.8-RC1
+	*/
+	$vars = array('message', 'bbcode_uid', 'match', 'replace');
```
Comment (marc1706) [inline phpBB/includes/functions_content.php]: ```$message``` is passed by reference. Please use an alternative variable for the event. Also, the match and replace will no longer properly work for 3.2.

File: phpBB/includes/functions_content.php (hunk 1 above)
Comment (rxu) [inline phpBB/includes/functions_content.php]: Well, someone probably would use custom preg_expression and related match set.

File: phpBB/includes/functions_content.php (hunk 2 above)
Comment (rxu) [inline phpBB/includes/functions_content.php]: To be honest, I have no idea what to do with 3.2 text parser here. That's a real pita in regard of BC.

**Comment by rxu** [comment]
Updated #4086.

**Pull request merged**